### PR TITLE
Fix resovle API bug

### PR DESCRIPTION
**Pull request by pulltheflower**
- Fix resolve API bug
- Add swagger API for resolve API

### CODE REVIEW

File: api/handler/model.go [hunk 1]
```diff
@@ -892,6 +892,30 @@ func (h *ModelHandler) SDKListFiles(ctx *gin.Context) {
...
+// @Param        namespace path string true "namespace"
+// @Param        name path string true "name"
+// @Param        file_path path string true "file_path"
+// @Param        ref query string false "ref"
+// @Success      200  {object}  types.Response{data=string} "OK"
+// @Failure      400  {object}  types.APIBadRequest "Bad request"
+// @Failure      500  {object}  types.APIInternalServerError "Internal server error"
+// @Router       /models/{namespace}/{name}/resolve/{file_path} [get]
+func (h *ModelHandler) ResolveDownload(ctx *gin.Context) {
+	h.handleDownload(ctx)
```
Comment (Rader) [inline api/handler/model.go:915]: call with param `isResolve`  = true

File: api/handler/model.go [hunk 2]
```diff
@@ -959,17 +974,36 @@ func (h *ModelHandler) HeadSDKDownload(ctx *gin.Context) {
...
+		if err != nil {
+			slog.Error("Failed to download model file", slog.Any("error", err))
+			httpbase.ServerError(ctx, err)
+			return
+		}
+	}
+}
+
+func isResolveAPI(ctx *gin.Context) bool {
+	return strings.EqualFold(ctx.FullPath(), "/api/v1/models/:namespace/:name/resolve/*file_path")
```
Comment (Rader) [inline api/handler/model.go:1008]: check url is a bad idea

Comment (pulltheflower) [reply api/handler/model.go]: Thanks for your review. Removed it.

File: api/handler/model.go [hunk 3]
```diff
@@ -910,46 +934,37 @@ func (h *ModelHandler) SDKDownload(ctx *gin.Context) {
...
 	if err != nil {
 		slog.Error("Bad request format", "error", err)
 		httpbase.BadRequest(ctx, err.Error())
 		return
 	}
+
 	filePath := ctx.Param("file_path")
 	filePath = convertFilePathFromRoute(filePath)
-	branch := ctx.Param("branch")
+	if isResolveAPI(ctx) {
```
[the remark this answers is not in the document]
Comment (pulltheflower) [reply api/handler/model.go]: Done.

**Review by Rader — APPROVED** [review]

**Pull request merged**